README.md

```diff
@@ -50,6 +50,28 @@ have dependencies on other algebras which must be implemented.
     - Two promises are equivalent when they yield equivalent values.
     - Two functions are equivalent if they yield equivalent outputs for equivalent inputs.
 
+## Type signature notation
+
+The type signature notation used in this document is described [here](https://sanctuary.js.org/#types). In summary:
+
+1. New types can be created via type constructors. Type constructors take one or more type arguments. For example, the type "array of strings" is written as `Array String`, and the type constructor `Array` can be defined as `Array :: a -> a`.
```

Comment:
> `Maybe` given a Type gives back a `Maybe Type`. `Array`, given a type gives back an `Array Type` not just a `Type`.

This is incorrect. An array of types looks like this:

```purescript
[$.Boolean, $.Number, $.String] :: Array Type
```

`$.Array($.Integer)`, on the other hand, represents a single type—`Array Integer`—rather than an array of types:

```purescript
$.Array :: Type -> Type
$.Array($.Integer) :: Type
```

It's important to keep in mind that the type of `$.Array($.Integer)` is *not* `Array Integer`. Thinking about the types of types is especially confusing in a language without types, where types—and the types of types—can only exist at the value level.